waspc/src/Wasp/AppSpec/App/Db.hs

```diff
@@ -3,34 +3,16 @@
 module Wasp.AppSpec.App.Db
   ( Db (..),
     DbSystem (..),
-    PrismaOptions (..),
-    PrismaDbExtension (..),
   )
 where
 
 import Data.Data (Data)
 import Wasp.AppSpec.ExtImport (ExtImport)
 
 data Db = Db
-  { system :: Maybe DbSystem,
-    seeds :: Maybe [ExtImport],
-    prisma :: Maybe PrismaOptions
+  { seeds :: Maybe [ExtImport]
   }
   deriving (Show, Eq, Data)
 
-data DbSystem = PostgreSQL | SQLite
-  deriving (Show, Eq, Data)
-
-data PrismaOptions = PrismaOptions
-  { clientPreviewFeatures :: Maybe [String],
-    dbExtensions :: Maybe [PrismaDbExtension]
-  }
-  deriving (Show, Eq, Data)
-
-data PrismaDbExtension = PrismaDbExtension
-  { name :: String,
-    map :: Maybe String,
-    version :: Maybe String,
-    schema :: Maybe String
-  }
+data DbSystem = PostgreSQL | SQLite | UnsupportedDbSystem String | MissingDbSystem
```

Comment:
I'm now throwing an error in the process of constructing the AppSpec if the db system is not valid. https://github.com/wasp-lang/wasp/pull/2035/commits/40d8da3baee83f6b479f822a67f7665ef63f0fcb#diff-887c7d76b4c6dcf05205fc8dfc861d241f7d2b75310ceb2c06cfdb21904337cdR47

I had to throw and not do the usual validation in the `Valid.hs` since the dev DB url depends on the DB system which would be we would need to construct the dev DB url based on invalid DB system and then throw a validation error regarding the DB system afterward which felt wrong to do.

Reply:
Hm got it. Ok, this is not so bad. We still have that `dbSystem` field which is duplicated compared to prismaSchema, which is not great but passable. Comment might help, saying it was extract from prisma schema.
Maybe we should be constructing db url also later then, what about that? Or actually, maybe we shouldn't construct it, but it should also be a getter?

Reply:
The dev DB url can stay as is since now the construction of the AppSpec only happens when the Prisma schema is validated 👍

Reply:
Awesome, everything falling into place now!

You can resolve.